### PR TITLE
Bump ring 0.17 in dev-deps, bump to latest

### PR DESCRIPTION
**Pull request by djc**
Let's see what the benchmark results look like.

### CODE REVIEW

File: examples/Cargo.toml [hunk 1]
```diff
@@ -24,4 +24,4 @@ webpki-roots = "=0.26.0-alpha.1"
 
 [dev-dependencies]
 regex = "1.9"
-ring = "0.16.20"
+ring = "0.17"
```
Comment (ctz) [inline examples/Cargo.toml:27]: AFAICT this and regex actually can be deleted, as neither are used in this crate

**Review by ctz — APPROVED** [review]

File: examples/Cargo.toml (hunk 1 above)
Comment (djc) [reply examples/Cargo.toml]: Ah, yes, did that.

**Comment by aochagavia** [comment]
As [discussed](https://discord.com/channels/976380008299917365/1015156984007381033/1161205005269864530) on Discord, it looks like there is a regression when GCC is used to compile ring's C code, but not when Clang is used. The bench runner is configured to use GCC, so it shows the regression.

I've tested using Clang 14 and there seem to be no regressions there (see [here](https://gist.github.com/aochagavia/231b903c834956837a9a7c4c79037bf2) for the details).

**Comment by MichaIng** [comment]
This is highly welcome, also since ring 0.17 adds support for RISC-V. But builds still fail since some seems to still keep ring 0.16 inside: https://github.com/rustls/rustls/blob/ac10422/Cargo.lock#L773

I do not see any other create adding it as dependency, so can this duplicate+old ring entry be removed from the `Cargo.lock`?

**Comment by djc** [comment]
Those are just dev-dependencies, it shouldn't affect anything if you just want to build the library.

**Comment by MichaIng** [comment]
I recognised just now that this change, and the one which moved `ring` from deps to dev-deps has not been released yet. I will test it with an 0.22 alpha version in our project.

**Pull request merged**